java-frontend/src/main/java/org/sonar/java/se/MethodYield.java

```diff
@@ -54,12 +59,17 @@ public String toString() {
     return "{params: " + Arrays.toString(parametersConstraints) + ", result: " + resultConstraint + " (" + resultIndex + "), exceptional: " + exception + "}";
   }
 
-  public Collection<ProgramState> statesAfterInvocation(List<SymbolicValue> invocationArguments, ProgramState programState, Supplier<SymbolicValue> svSupplier) {
+  public Collection<ProgramState> statesAfterInvocation(List<SymbolicValue> invocationArguments, List<Type> invocationTypes, ProgramState programState,
+    Supplier<SymbolicValue> svSupplier) {
     Set<ProgramState> results = new LinkedHashSet<>();
     for (int index = 0; index < invocationArguments.size(); index++) {
-      // FIXME : varargs method should be handled
       SymbolicValue invokedArg = invocationArguments.get(index);
-      Constraint constraint = parametersConstraints[Math.min(index, parametersConstraints.length - 1)];
+      Constraint constraint = null;
+      if (!varArgs || varArgsCalledWithOneArrayParameter(index, invocationTypes)) {
+        constraint = parametersConstraints[index];
+      } else {
```

Comment:
I am not a big fan of an else for the sole purpose of a comment. Rephrase it to make it applicable for the condition ?